src/events/message_delete.rs

```diff
@@ -68,19 +68,50 @@ pub async fn message_delete(
         .map(|entry| entry.user_id)
         .and_then(|deletor| audit_logs.users.iter().find(|usr| usr.id == deletor));
 
+    let image = if let Some(_image) = msg
+        .attachments
+        .iter()
+        .filter(|a| a.height.is_some() && a.height.unwrap() > 0)
+        .next()
+    {
+        let image_path = std::path::Path::new("cache").join(msg.id.to_string());
+        if image_path.exists() {
+            Some(tokio::fs::File::open(image_path).await.unwrap())
+        } else {
+            None
+        }
+    } else {
+        None
+    };
+    let image = image.as_ref();
+
     config
         .channel_bot_messages
-        .send_embed(&ctx, |e| {
-            e.author(|a| a.name("Message Deleted").icon_url(msg.author.face()));
-            e.title(msg.author.name_with_disc_and_id());
-            e.description(format!("{}\n\n{}", msg.content, msg.to_context_link()));
-            e.footer(|f| {
-                f.text(format!(
-                    "#{}{}",
-                    channel_name,
-                    deletor.map_or(Default::default(), |x| format!(", deleted by {}", x.tag()))
-                ))
-            });
+        .send_message(&ctx.http, |m| {
+            dbg!(&image);
+            if let Some(image) = image {
+                let attachment = serenity::http::AttachmentType::File {
+                    file: image,
+                    filename: "image.jpg".to_string(),
+                };
+                m.add_file(attachment);
+            }
+
+            m.embed(|e| {
+                e.author(|a| a.name("Message Deleted").icon_url(msg.author.face()));
+                e.title(msg.author.name_with_disc_and_id());
+                e.description(format!("{}\n\n{}", msg.content, msg.to_context_link()));
+                if image.is_some() {
+                    e.attachment("image.jpg");
+                }
```

Comment:
Using [CreateEmbed::image](https://docs.rs/serenity/0.10.4/serenity/builder/struct.CreateEmbed.html#method.image) seems cleaner than manually adding the attachment and in this multi-step process, no? or am I missing something

Reply:
> than manually adding the attachment

Multiple images, maybe?

Reply:
technically possible, although idk if a bot can even do that.,... and currently the code wouldn't have done multiple images anyways.

Reply:
::image only supports passing a URL, you need to .add_file the file, e.attachment already calls image with `attachment://{filename}`

Reply:
> technically possible, although idk if a bot can even do that

Fairly certain that current-trup already does.

Reply:
Well, it used to, at least.

I did a quick test and it doesn't seem to do so anymore, maybe i attached too many?  maybe the "small" refactor/rewrite broke it?  idk

Reply:
maybe there wasn't enough time between the things:tm:?  no idea, too lazy to test.